packages/merge-styles/src/Stylesheet.ts

```diff
@@ -252,19 +309,21 @@ export class Stylesheet {
       this._rules.push(rule);
     }
 
+    // eslint-disable-next-line deprecation/deprecation
     if (this._config.onInsertRule) {
+      // eslint-disable-next-line deprecation/deprecation
       this._config.onInsertRule(rule);
     }
+
+    this._onInsertRuleCallbacks.forEach(callback => callback());
   }
 
   /**
    * Gets all rules registered with the stylesheet; only valid when
    * using InsertionMode.none.
    */
   public getRules(includePreservedRules?: boolean): string {
-    return (
-      (includePreservedRules ? this._preservedRules.join('') : '') + this._rules.join('') + this._rulesToInsert.join('')
-    );
+    return (includePreservedRules ? this._preservedRules.join('') : '') + this._rules.join('');
```

Comment:
`_rulesToInsert` didn't do anything and it was more of an artefact of previous implementation. If some users set it directly on the instance this will break for them, but that behaviour was never supported in the first place.

Reply:
@dzearing This behaviour creates a duplicity of preserved rules, are we okay with that? It shouldn't affect anything other than bundle size as registering 10 rules all as preserved would collect them from the `_preservedRules` array as well as in `_rules` resulting in twice as long string due to duplication.

Reply:
You can see the behaviour captured here: https://github.com/microsoft/fluentui/pull/19888/files#diff-0ffab654f1856f5740c61ba548c5256132f11c889bd7da79e2cee8ed6f464f0dR386

If you are observing a single instance of every rule, that could be caused by:
- no rules were specified as preserved
- all rules were specified as preserved and the stylesheet was reset before getting them